crates/core/src/db/relational_db.rs

```diff
@@ -60,6 +65,7 @@ pub struct RelationalDB {
     // TODO(cloutiertyler): This should not be public
     pub(crate) inner: Locking,
     durability: Option<Arc<dyn Durability<TxData = Txdata>>>,
+    snapshots: Option<Arc<SnapshotWorker>>,
```

Comment:
```suggestion
    snapshotter: Option<Arc<SnapshotWorker>>,
```
perhaps?

I wonder if this should be owned by the `RelationalDB` or if it would just be better to have this be something held by the `Host` or `DatabaseInstanceContext` and to put `maybe_do_snapshot` in there as well. Although I haven't through through exactly who should call `maybe_do_snapshot`

Reply:
I think it makes the most sense to store in the `RelationalDB` because:
- It acts a lot like the `Durability`, which is also owned by the `RelationalDB`.
- Opportunities to call `maybe_do_snapshot` become increasingly tenuous as you get more abstracted than this.
- I would prefer not to leak the `CommittedState`, or even the `Locking`, any further than this.

Reply:
I've renamed all the `snapshots` vars and members to either `snapshot_worker` or `snapshot_repo` as appropriate.